sklearn/metrics/_pairwise_distances_reduction/_base.pyx.tp

```diff
@@ -1,3 +1,18 @@
+{{py:
+
+implementation_specific_values = [
+    # Values are the following ones:
+    #
+    #       name_suffix, upcast_to_float64, INPUT_DTYPE_t, INPUT_DTYPE
+    #
+    # We also use the float64 dtype and C-type names as defined in
+    # `sklearn.utils._typedefs` to maintain consistency.
+    #
+    ('64', False, 'DTYPE_t', 'DTYPE'),
+    ('32', True, 'cnp.float32_t', 'np.float32')
```

Comment:
We only need `name_suffix`, `INPUT_DTYPE_t` and `INPUT_DTYPE` to generate this template files sources.